### PR TITLE
Generalize uvisor lib

### PR DESCRIPTION
**Pull request by AlessandroA**
This PR implements the changes from https://github.com/ARMmbed/uvisor/pull/200.

* The `unsupported.*` files have been removed (they now live in ARMmbed/uvisor).
* The uVisor-specific symbols `UVISOR_PRESENT` and `ARCH_MPU_*` have been translated using yotta-specific symbols.

@meriac @Patater @niklas-arm

### CODE REVIEW

File: uvisor-lib/uvisor-lib.h [hunk 1]
```diff
@@ -17,38 +17,31 @@
...
+#endif /* defined(TARGET_LIKE_ARMV7_M) */
 
 #else /* defined(YOTTA_CFG_UVISOR_PRESENT) && YOTTA_CFG_UVISOR_PRESENT == 1 */
 
-#include "uvisor-lib/unsupported.h"
+/* There is no need to specify any other symbol if uVisor is unsupported. */
+#define UVISOR_PRESENT 0
 
-#endif /* defined(YOTTA_CFG_UVISOR_PRESENT) && YOTTA_CFG_UVISOR_PRESENT == 1 */
+#endif /* define(YOTTA_CFG_UVISOR_PRESENT) && YOTTA_CFG_UVISOR_PRESENT == 1 */
```
Comment (Patater) [inline uvisor-lib/uvisor-lib.h]: This should stay as "defined".

File: uvisor-lib/uvisor-lib.h [hunk 2]
```diff
@@ -17,38 +17,31 @@
...
 
-#include "uvisor/api/inc/debug_exports.h"
-#include "uvisor/api/inc/halt_exports.h"
-#include "uvisor/api/inc/svc_exports.h"
-#include "uvisor/api/inc/svc_gw_exports.h"
-#include "uvisor/api/inc/unvic_exports.h"
-#include "uvisor/api/inc/uvisor_exports.h"
-#include "uvisor/api/inc/vmpu_exports.h"
+#include "uvisor/api/inc/uvisor-lib.h"
```
Comment (Patater) [inline uvisor-lib/uvisor-lib.h]: This is currently in "Translate yotta-specific symbols to uVisor ones" commit, but it should be in a different commit, as it doesn't translate any yotta-specific symbols to uVisor ones. It might fit better in "Remove outdated redirection header files".

**Comment by Patater** [comment]
What's the main benefit to moving "unsupported.c" out of uvisor-lib?

File: uvisor-lib/uvisor-lib.h (hunk 2 above)
Comment (AlessandroA) [inline uvisor-lib/uvisor-lib.h]: But the commit message says "With this commit uvisor-lib/uvisor-lib.h becomes a wrapper to uvisor/api/inc/uvisor-lib.h", which is what I'm doing here. The purpose of the commit is to remove all includes and add a single include preceded by yotta -> uvisor translation. What about a commit title "Turn uvisor-lib.h into wrapper for yotta->uvisor translation". ?

**Comment by AlessandroA** [comment]
That fallback for unsupported platforms is rather a uVisor-specific feature, not an OS one, hence I think it should live in uVisor. Operating systems that are interested in giving this fallback support will make sure to implement a selection logic of the form:

```
if supported
    link target.a
else
   build unsupported.c
endif
```

File: uvisor-lib/uvisor-lib.h (hunk 2 above)
Comment (Patater) [inline uvisor-lib/uvisor-lib.h]: Does the build work after "Remove outdated redirection header files" without also modifying this file to replace all includes with a single include? If not, then this probably has to be done with that commit. It seems to me like the removal of the redirection headers is what is driving this change.

However, it could be the other way around. Perhaps your desire to "turn uvisor-lib.h into a wrapper for yotta->uvisor translation" is what drove the removal of the redirection includes. In which case, I'd put that removal in the "Turn uvisor-lib.h into wrapper for yotta->uvisor translation" commit.

**Comment by AlessandroA** [comment]
Note: The failure in the CI comes from the missing `unssuported.c` file, which is expected to come from AMRmbed/uvisor once the [related PR](https://github.com/ARMmbed/uvisor/pull/200) is merged.

**Comment by Patater** [comment]
The reason for the move makes sense. It'd be good to document it in one of the commit messages.
